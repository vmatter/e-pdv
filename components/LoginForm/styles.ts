import styled from '@emotion/styled';

export const FormWrapper = styled.section`
  max-width: 480px;
  width: 100%;
  height: 100vh;
  padding: 0 24px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const InputWrapper = styled.form`
  display: flex;
  flex-direction: column;
  padding: 12px;
  gap: 12px;
`;
