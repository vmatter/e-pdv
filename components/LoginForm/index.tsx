import { Button, TextField, Typography } from '@material-ui/core';
import { useRouter } from 'next/router';
import { useState, useEffect } from 'react';
import { FormWrapper, InputWrapper } from './styles';

const LoginForm = () => {
  const router = useRouter();
  const [loggedIn, setLoggedIn] = useState(false);
  const [loading, setLoading] = useState(false);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {
    loggedIn && router.push('/');
  }, [loggedIn]);

  const onSubmit = async (e: any) => {
    e?.preventDefault();

    setLoading(true);
    console.log(`email`, email);
    console.log(`password`, password);

    setTimeout(() => {
      setLoading(false);
      setLoggedIn(true);
    }, 2000);
  };

  return (
    <FormWrapper>
      <Typography variant="h5" component="h2">
        e-PDV Sign In
      </Typography>
      <InputWrapper>
        <TextField
          id="input-email"
          data-testid="input-email"
          variant="outlined"
          type="text"
          name="username"
          label="username"
          onChange={(e: any) => {
            setEmail(e.target.value);
          }}
        />
        <TextField
          id="input-password"
          data-testid="input-password"
          variant="outlined"
          type="password"
          name="password"
          label="password"
          onChange={(e: any) => {
            setPassword(e.target.value);
          }}
        />
        <Button
          id="button-login"
          variant="contained"
          color="primary"
          disabled={loading}
          onClick={onSubmit}
        >
          Login
        </Button>
      </InputWrapper>
    </FormWrapper>
  );
};

export default LoginForm;
