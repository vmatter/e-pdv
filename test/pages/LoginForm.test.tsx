import React from 'react';
import { render } from '../testUtils';
import LoginForm from '../../components/LoginForm';

jest.mock('next/router');

describe('LoginForm', () => {
  it('matches snapshot', () => {
    const { container } = render(<LoginForm />, {});
    expect(container.firstChild).toMatchSnapshot();
  });
});
