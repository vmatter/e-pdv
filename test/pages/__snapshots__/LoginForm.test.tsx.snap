// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-y9ds5f-Wrapper e16kqbw65"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="css-5m6oo8-HeaderWrapper e16kqbw64"
      >
        <div
          class="css-sdubf6-FormHeader e16kqbw63"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-ggtcmo-MuiTypography-root"
          >
            Log in
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1m0ae9b-MuiTypography-root"
          >
            Fa√ßa log in na plataforma e-PDV
          </p>
        </div>
        <a
          class="css-5dglsj-LogoWrapper e16kqbw61"
        >
          <img
            class="css-1l9u5cb-LogoImg e16kqbw60"
            src="/pdv-logo.png"
          />
        </a>
      </div>
      <form
        autocomplete="off"
        class="css-1dpfc4n-InputWrapper e16kqbw62"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="input-email"
            id="input-email-label"
          >
            email
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="input-email"
              id="input-email"
              name="email"
              placeholder="ex: johndoe@gmail.com"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-16yrtjo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="input-password"
            id="input-password-label"
          >
            senha
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1g6hcx9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-weuz2y-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="input-password"
              id="input-password"
              name="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  senha
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1iqv7g7-MuiButtonBase-root-MuiButton-root"
          data-testid="button-login"
          id="button-login"
          tabindex="0"
        >
          <span
            class="MuiButton-label css-8xplcm-MuiButton-label"
          >
            Log In
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
